azurelinuxagent/common/protocol/restapi.py

```diff
@@ -73,6 +73,8 @@ def __init__(self, name, version=None):
         self.name = name
         # This is the Requested version as specified by the Goal State, it defaults to 0.0.0.0 if not specified in GS
         self.requested_version_string = VERSION_0 if version is None else version
+        # Set to False if the version from rsm property is not specified in the GS
+        self.is_version_from_rsm = False
```

Comment:
having is_version_from_rsm and is_requested_version_specified seems redundant/overcomplicated. can we simplify that?